core/kamon-core/src/main/scala/kamon/package.scala

```diff
@@ -15,14 +15,25 @@
  */
 
 import java.util.concurrent.atomic.AtomicLong
-import java.util.concurrent.{Executors, ThreadFactory}
-
+import java.util.concurrent.{Executors, ScheduledExecutorService, ScheduledThreadPoolExecutor, ThreadFactory, TimeUnit}
 import com.typesafe.config.{Config, ConfigUtil}
 
 import scala.collection.concurrent.TrieMap
 
 package object kamon {
 
+  def newScheduledThreadPool(corePoolSize: Int, threadFactory: ThreadFactory): ScheduledExecutorService = {
```

Comment:
@ivantopo this will make the threads stop if there is no more work in the queues. I got this from https://github.com/typelevel/cats-effect/blob/series/2.x/core/jvm/src/main/scala/cats/effect/internals/IOTimer.scala#L72-L98.

Do you think we should make this configurable instead?

Reply:
I wouldn't make it configurable unless there was an effect to end users. For now, leave it as it is.